storageClient/storageClient_test.go

```diff
@@ -330,3 +333,186 @@ func TestReadGroupWithValidInputSucceeds(t *testing.T) {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", newGroup.Name, response.Name, newGroup)
+	}
+	response, err = queenClient.CreateGroup(testCtx, newGroup2)
+	if err != nil {
+		t.Fatalf("Failed to create group \n Group( %+v) \n error %+v", newGroup2, err)
+	}
+	if response.Name != newGroup2.Name {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", newGroup2.Name, response.Name, newGroup2)
+	}
+	listRequest := ListGroupsRequest{
+		ClientID:  uuid.MustParse(queenClient.ClientID),
+		NextToken: 0,
+		Max:       10,
+	}
+	responseList, err := queenClient.ListGroups(testCtx, listRequest)
+	if err != nil {
+		t.Fatalf("Failed to list Group: Response( %+v) \n error %+v", responseList, err)
+	}
+}
+
+func TestListGroupsWithQueenClientAllAccountReturnsSuccess(t *testing.T) {
+	registrationClient := accountClient.New(e3dbClients.ClientConfig{Host: cyclopsServiceHost})
+	queenClientInfo, createAccountResponse, err := e3dbTest.MakeE3DBAccount(t, &registrationClient, uuid.New().String(), cyclopsServiceHost)
+	if err != nil {
+		t.Fatalf("Error %s making new account", err)
+	}
+	queenClientInfo.Host = cyclopsServiceHost
+	accountToken := createAccountResponse.AccountServiceToken
+	queenClient := New(queenClientInfo)
+	queenClient2 := New(queenClientInfo)
+	encryptionKeyPair, err := e3dbClients.GenerateKeyPair()
+	if err != nil {
+		t.Fatalf("error %s creating account registration token using %+v %+v", err, queenClient, accountToken)
+	}
+	// Create Another Accouunt under the queen client to list groups, Each create a group
+	group := CreateGroupRequest{
+		Name:              "TestGroup1" + uuid.New().String(),
+		PublicKey:         encryptionKeyPair.Public.Material,
+		EncryptedGroupKey: encryptionKeyPair.Private.Material,
+	}
+	response, err := queenClient2.CreateGroup(testCtx, group)
+	if err != nil {
+		t.Fatalf("Failed to create group \n Group( %+v) \n error %+v", group, err)
+	}
+	if response.Name != group.Name {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", group.Name, response.Name, group)
+	}
+	group = CreateGroupRequest{
+		Name:              "TestGroup1" + uuid.New().String(),
+		PublicKey:         encryptionKeyPair.Public.Material,
+		EncryptedGroupKey: encryptionKeyPair.Private.Material,
+	}
+	response, err = queenClient.CreateGroup(testCtx, group)
+	if err != nil {
+		t.Fatalf("Failed to create group \n Group( %+v) \n error %+v", group, err)
+	}
+	if response.Name != group.Name {
+		t.Fatalf("Group name (%+v) passed in, does not match Group name (%+v) inserted for Group( %+v) \n", group.Name, response.Name, group)
+	}
+	// List The Groups Under the Account
+	listRequest := ListGroupsRequest{}
+	responseList, err := queenClient.ListGroups(testCtx, listRequest)
+	if err != nil {
+		t.Fatalf("Failed to list Group: Response( %+v) \n error %+v", responseList, err)
+	}
+	t.Log(responseList)
```

Comment:
this can be removed, test that the response contains the group created on line 456